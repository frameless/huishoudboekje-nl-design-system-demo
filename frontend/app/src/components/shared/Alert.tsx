import {AlertDialog, AlertDialogBody, AlertDialogContent, AlertDialogFooter, AlertDialogHeader, AlertDialogOverlay, Button} from "@chakra-ui/react";
import React, {useRef} from "react";
import {useTranslation} from "react-i18next";

const Alert = ({title, children, confirmButton, cancelButton = true, onClose}) => {
	const cancelDeleteRef = useRef(null);
	const {t} = useTranslation();

	return (
		<AlertDialog isOpen={true} leastDestructiveRef={cancelDeleteRef} onClose={onClose}>
			<AlertDialogOverlay />
			<AlertDialogContent>
				<AlertDialogHeader fontSize="lg" fontWeight="bold">{title}</AlertDialogHeader>
				<AlertDialogBody>{children}</AlertDialogBody>
				<AlertDialogFooter>
					{cancelButton && <Button ref={cancelDeleteRef} onClick={onClose} data-cy={"inModal"}>{t("global.actions.cancel")}</Button>}
					{confirmButton}
				</AlertDialogFooter>
			</AlertDialogContent>
		</AlertDialog>
	);
};

export default Alert;