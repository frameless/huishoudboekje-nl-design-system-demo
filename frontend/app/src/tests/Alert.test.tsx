import {act} from "@testing-library/react";
import React from "react";
import {render, unmountComponentAtNode} from "react-dom";
import Alert from "../components/shared/Alert";
import {Button} from "@chakra-ui/react";

jest.mock("react-i18next", () => require("./utils/mock-hooks").reactI18NextMock());

let container: HTMLDivElement | null = null;

beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container!);
    container!.remove();
    container = null;
});

it("show alert with cancel button", () => {

    act(() => {
        render(
            <Alert
                title={"Burger verwijderen uit huishouden"}
                confirmButton={<Button colorScheme={"red"} ml={3}>Verwijderen</Button>}
                cancelButton={true}
                onClose={() => void (0)}
            >
                Weet je zeker dat je Chris de Burg wil verwijderen uit het huishouden de Jager-de Burg?
            </Alert>
            , container);
    });

    const html = container!.innerHTML;
    expect(html).not.toBeNull();
});

it("show alert without cancel button", () => {

    act(() => {
        render(
            <Alert
                title={"Burger verwijderen uit huishouden"}
                confirmButton={<Button colorScheme={"red"} ml={3}>Verwijderen</Button>}
                cancelButton={false}
                onClose={() => void (0)}
            >
                Weet je zeker dat je Chris de Burg wil verwijderen uit het huishouden de Jager-de Burg?
            </Alert>
            , container);
    });

    const html = container!.innerHTML;
    expect(html).not.toBeNull();
});
