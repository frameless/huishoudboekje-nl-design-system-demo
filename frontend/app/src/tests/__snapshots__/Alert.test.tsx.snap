// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert CloseButton and cancelButton are working 1`] = `
<section
  aria-describedby="chakra-modal--body-:r5:"
  aria-labelledby="chakra-modal--header-:r5:"
  aria-modal="true"
  class="chakra-modal__content css-6yiygt"
  id="chakra-modal-:r5:"
  role="alertdialog"
  style="opacity: 0; transform: scale(0.95) translateZ(0);"
  tabindex="-1"
>
  <header
    class="chakra-modal__header css-1jynom4"
    id="chakra-modal--header-:r5:"
  >
    Burger verwijderen uit huishouden
  </header>
  <button
    aria-label="Close"
    class="chakra-modal__close-btn css-6sdaxy"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="chakra-icon css-onkibi"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
        fill="currentColor"
      />
    </svg>
  </button>
  <div
    class="chakra-modal__body css-0"
    id="chakra-modal--body-:r5:"
  >
    Weet je zeker dat je Chris de Burg wil verwijderen uit het huishouden de Jager-de Burg?
  </div>
  <footer
    class="chakra-modal__footer css-192qrng"
  >
    <button
      class="chakra-button css-4xx2wk"
      type="button"
    >
      global.actions.cancel
    </button>
    <button
      class="chakra-button css-1i56vmk"
      type="button"
    >
      Verwijderen
    </button>
  </footer>
</section>
`;

exports[`Alert Shows the Alert 1`] = `
<section
  aria-describedby="chakra-modal--body-:r0:"
  aria-labelledby="chakra-modal--header-:r0:"
  aria-modal="true"
  class="chakra-modal__content css-6yiygt"
  id="chakra-modal-:r0:"
  role="alertdialog"
  style="opacity: 0; transform: scale(0.95) translateZ(0);"
  tabindex="-1"
>
  <header
    class="chakra-modal__header css-1jynom4"
    id="chakra-modal--header-:r0:"
  >
    Burger verwijderen uit huishouden
  </header>
  <button
    aria-label="Close"
    class="chakra-modal__close-btn css-6sdaxy"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="chakra-icon css-onkibi"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
        fill="currentColor"
      />
    </svg>
  </button>
  <div
    class="chakra-modal__body css-0"
    id="chakra-modal--body-:r0:"
  >
    Weet je zeker dat je Chris de Burg wil verwijderen uit het huishouden de Jager-de Burg?
  </div>
  <footer
    class="chakra-modal__footer css-192qrng"
  >
    <button
      class="chakra-button css-4xx2wk"
      type="button"
    >
      global.actions.cancel
    </button>
  </footer>
</section>
`;
